projects/gateway2/translator/plugins/listeneroptions/listener_options_plugin.go

```diff
@@ -47,7 +49,11 @@ func (p *plugin) ApplyListenerPlugin(
 	// use the first option (highest in priority)
 	// see for more context: https://github.com/solo-io/solo-projects/issues/6313
 	optToUse := attachedOptions[0]
-	outListener.Options = optToUse.Spec.GetOptions()
+	if outListener.GetOptions() != nil {
```

Comment:
We have some very similar code for our other policies defined here: https://github.com/solo-io/gloo/blob/d4fa2772b13d3325011e4fcaf08d53128cf5659b/projects/gloo/pkg/utils/merge.go#L83. I think it would make sense to co-locate all our merge code to avoid the implementations drifting, if possible. What do you think?

Reply:
I guess I could write a manual one.. but I think the proto merge semantics make sense here. ListenerOptions has lists and maps, i'd imagine we want those to merge although the specific bug I care about only needs the top level fields to be merged.

Reply:
okay updated it to do this